### PR TITLE
Only show 418 debug info in when `VERBOSE_ERROR_PAGES === true`

### PR DESCRIPTION
**Pull request by JoeChapman**
- wrap debug in ~~ENV~~ $VERBOSE_ERROR_PAGES  if block
- update readme to reflect change and provide guidance

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ pages. - Please read [the documentation for that image].
 
 **Warning:** This also adds debugging information to 418 (Teapot) pages.
              You might not want this if you want to take the small
-             advantage of 'security through obscurity'. Feel free to
-             mount in your own SSI 'partial' to remove this information.
+             advantage of 'security through obscurity'. Set an ENV to
+             'production' so the degug information is no rendered.
```
Comment (lennym) [inline README.md]: `degug`

Comment (lennym) [inline README.md]: > no rendered

Comment (JoeChapman) [inline README.md]: 🤦‍♂️

File: html/partials/418-request-denied.shtml [hunk 2]
```diff
@@ -3,9 +3,11 @@
   made a mistake please raise a support ticket with the following
   information:
```
Comment (daniel-ac-martin) [inline html/partials/418-request-denied.shtml]: This text would need to change when we don't show the debug info.

**Comment by JoeChapman** [comment]
@daniel-ac-martin is this better?

File: html/partials/418-request-denied.shtml [hunk 3]
```diff
@@ -1,11 +1,17 @@
 <p>
-  Sorry, we are refusing to process your request. If you believe we have
-  made a mistake please raise a support ticket with the following
-  information:
+  <!--#if expression="$VERBOSE_ERROR_PAGES" -->
```
Comment (daniel-ac-martin) [inline html/partials/418-request-denied.shtml]: This doesn't work.

File: html/partials/418-request-denied.shtml [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
+    Sorry, we are refusing to process your request. If you believe we have
+    made a mistake please raise a support ticket with the following
+    information:
+  <!--#elif -->
+    We have been unable to process the information you tried to submit.
+  <!--#endif -->
 </p>
+<!--#if expression="$VERBOSE_ERROR_PAGES" -->
```
Comment (daniel-ac-martin) [inline html/partials/418-request-denied.shtml]: This doesn't work.

**Comment by JoeChapman** [comment]
@daniel-ac-martin I was using the wrong templating attribute `expression` and not `expr`.

**Comment by JoeChapman** [comment]
@easternbloc please review

File: html/partials/418-request-denied.shtml [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
-  information:
+  <!--#if expr="$VERBOSE_ERROR_PAGES" -->
+    Sorry, we are refusing to process your request. If you believe we have
+    made a mistake please raise a support ticket with the following
+    information:
+  <!--#elif -->
+    We have been unable to process the information you tried to submit.
+  <!--#endif -->
 </p>
+<!--#if expr="$VERBOSE_ERROR_PAGES" -->
```
Comment (easternbloc) [inline html/partials/418-request-denied.shtml]: you could have probably rolled this into the first if but meh it's good enough

**Comment by daniel-ac-martin** [comment]
Sorry guys, this still looks broken to me.
```bash
$ docker build -t nginx-proxy-govuk .
$ docker run -e "PROXY_SERVICE_HOST=http://www.google.co.uk" -e "PROXY_SERVICE_PORT=80" -p "10443:10443" nginx-proxy-govuk
```
Then visiting a denied page such as, `https://localhost:10443/'`, gives me something like this:
screenshot from 2017-02-13 14-49-05

**Comment by JoeChapman** [comment]
@daniel-ac-martin thanks for testing. I have updated the commit. However, I've tested this and it  does not honour the env var `VERBOSE_ERROR_PAGES` and never renders the debug info. After some reading I can't find a way to compile SSI with custom env vars. Dan, is there an obvious way I cannot see?

**Comment by daniel-ac-martin** [comment]
@JoeChapman: It's been a long time since I've worked on this. I vaguely recall it being a pain. I think it must be possible though. I think you will need to manually pass the variable through in the nginx config. (i.e. you'd probably need to make a PR on the parent image.) I'd look for the code where I set the debug info variable up as an example.

Nag me tomorrow and I'll see if I can find it!

**Comment by JoeChapman** [comment]
@daniel-ac-martin I followed your recommendation as per your last comment but could not achieve the desired affect. Do you have any good advice?

**Comment by JoeChapman** [comment]
@daniel-ac-martin I've not made any changes, but this appears to be working now.

image

**Comment by daniel-ac-martin** [comment]
@JoeChapman: What about if the flag IS set?

**Comment by daniel-ac-martin** [comment]
I've just tested this and it looks like the else clause always triggers.

**Comment by daniel-ac-martin** [comment]
Upgrading the base image should now make this functionality possible.

File: README.md [hunk 6]
```diff
@@ -5,10 +5,20 @@
...
-             mount in your own SSI 'partial' to remove this information.
+## Debugging
+To have the template render with more debugging information set `VERBOSE_ERROR_PAGES` to `TRUE`.
+
+In the terminal
+```
+docker run -e "VERBOSE_ERROR_PAGES=TRUE" -p "10443:10443" nginx-proxy-govuk
+```
+
+In `yaml` file
```
Comment (daniel-ac-martin) [inline README.md]: 'In a Kubernetes config file'

But I think it's probably best to remove this as the image isn't really specific to k8s.

File: html/template.shtml [hunk 7]
```diff
@@ -88,10 +88,12 @@
...
-          Generated at <!--#config timefmt="%H:%M:%S %Z" --><!--#echo var="date_local" -->
-          on <!--#config timefmt="%A, %d %B %Y" --><!--#echo var="date_local" -->.
-        </p>
+        <!--#if expr="$verbose_error_pages = TRUE" -->
+          <p style="color: #6F777B;">
+            Generated at <!--#config timefmt="%H:%M:%S %Z" --><!--#echo var="date_local" -->
+            on <!--#config timefmt="%A, %d %B %Y" --><!--#echo var="date_local" -->.
+          </p>
+        <!--#endif -->
       </div>
```
Comment (daniel-ac-martin) [inline html/template.shtml]: Sure you want to remove this?

**Comment by daniel-ac-martin** [comment]
Otherwise, LGTM. (Still need to test.)

File: html/template.shtml (hunk 7 above)
Comment (JoeChapman) [reply html/template.shtml]: if non-debug yes, my user/content researcher/designer wants it gone

**Comment by JoeChapman** [comment]
Hi @daniel-ac-martin have you found time to test this yet?

**Comment by daniel-ac-martin** [comment]
@JoeChapman: Yo! - #14

**Comment by JoeChapman** [comment]
@daniel-ac-martin done done done

**Pull request merged**